js/siteorigin-panels/view/widget.js

```diff
@@ -20,8 +20,13 @@ module.exports = Backbone.View.extend( {
 	 * Initialize the widget
 	 */
 	initialize: function () {
-		this.model.on( 'user_edit', this.editHandler, this );				 // When a user wants to edit the widget model
-		this.model.on( 'user_duplicate', this.duplicateHandler, this );	   // When a user wants to duplicate the widget model
+		// When a user wants to edit the widget model
+		this.model.on( 'user_edit', this.editHandler, this );
+		// When a user wants to duplicate the widget model
+		this.model.on( 'user_duplicate', this.duplicateHandler, this );
+		// When the user created a new widget through a direct action
+		this.model.on( 'user_created_new', this.createdNewHandler, this );
```

Comment:
We should be consistent with tenses when triggering events. E.g. 'user_duplicate' should be triggered just before the duplication event is about to take place, where `user_created_new` should be triggered after the event has taken place.

** I just looked a bit closer at the code and realized these events are being used differently, which is confusing. `user_edit` and `user_duplicate` are actually user intentions? If so they should be triggered on views, instead of models... I think. 😕

Reply:
Agree with keeping tense the same. Will change that.

It's been a while, so I can't quite remember why we're triggering actions on the models. I know it had to do with a combination of factors like:

* Only having access to the model, and not the view in certain instances
* Needing to know the difference between user triggered and automatic events
* Overall inexperience with Backbone when I wrote some of this code :P

Will take a deep dive and see how much we can rearchitect this code to make it more logical and consistent. We can save that as a little project for the 2.7 update.